selendroid-standalone/src/main/java/io/selendroid/standalone/server/grid/SelfRegisteringRemote.java

```diff
@@ -136,7 +146,7 @@ private JSONObject getConfiguration() throws JSONException {
       configuration.put("proxy", "org.openqa.grid.selenium.proxy.DefaultRemoteProxy");
     }
     configuration.put("role", "node");
-    configuration.put("registerCycle", 5000);
+    configuration.put("registerCycle", config.getRegisterCycle());
```

Comment:
@asm89, the **registerCycle** value is time in milliseconds between recurrences of following operation:
 * check if already registered at hub. The check will return false when not registered. For example: a grid hub was restarted unexpectedly.
 * if the check above returned false, then perform registration at grid hub.

Originally in grid sources - [SelfRegisteringRemote.java](https://code.google.com/p/selenium/source/browse/java/server/src/org/openqa/grid/internal/utils/SelfRegisteringRemote.java), line 210:
`        Thread.sleep(registerCycleInterval);`

The default value for **registerCycleInterval** in code is set to **0** (i.e. recurrence is disabled)
[SelfRegisteringRemote.java](https://code.google.com/p/selenium/source/browse/java/server/src/org/openqa/grid/internal/utils/SelfRegisteringRemote.java), line 191:
`       final int registerCycleInterval = nodeConfig.getConfigAsInt(RegistrationRequest.REGISTER_CYCLE, 0);`


@asm89 and @koichirok, the value of 5000ms for **registerCycle** was set by me earlier, I've found it somewhere in sources as a default value.

But later experiments showed that when having a farm of over 100 of selendroid nodes with 4-6 emulators each will produce too much HTTP traffic just for the registration checks and result in flooding network and unnecessary load for grid hub.
So I've found a 60k milliseconds is a quite reasonable interval for checks. I think it could be set as high as equivalent of 5 minutes, because grid hub restarts happen not that often. But that's about personal choice. 

I'd like to suggest to set the default value to 0 as it will be inline with @asm89 point above about disabling the recurrence behavior and with original grid's sources [SelfRegisteringRemote.java](https://code.google.com/p/selenium/source/browse/java/server/src/org/openqa/grid/internal/utils/SelfRegisteringRemote.java), line 191.
`       final int registerCycleInterval = nodeConfig.getConfigAsInt(RegistrationRequest.REGISTER_CYCLE, 0);`

What is your opinion?

P.S.: original Selenium's references that might be of interest
[GridParameters.properties](https://code.google.com/p/selenium/source/browse/java/server/src/org/openqa/grid/common/defaults/GridParameters.properties) and [Grid2 wiki](https://code.google.com/p/selenium/wiki/Grid2#Optional_parameters)

Also, originally, in code design of [SelfRegisteringRemote.java](https://code.google.com/p/selenium/source/browse/java/server/src/org/openqa/grid/internal/utils/SelfRegisteringRemote.java) authors decided to use Thread.sleep in a while-true cycle to perform recurrence. I'd think that using Java API's  Timer and TimerTask would fit better.